test/e2e/e2e_test.go

```diff
@@ -1653,3 +1653,102 @@ func getNodePodCIDR(nodeName string) (string, error) {
 	}
 	return "", fmt.Errorf("could not parse annotation %q", annotation)
 }
+
+func checkPodSleep(f *framework.Framework, nodeName, podName string, errChan chan error) {
+	defer GinkgoRecover()
+
+	contName := fmt.Sprintf("%s-container", podName)
+
+	command := []string{
+		"bash", "-c",
+		"sleep 10",
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   framework.AgnHostImage,
+					Command: command,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	_, err := podClient.Create(pod)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	err = e2epod.WaitForPodNotPending(f.ClientSet, f.Namespace.Name, podName)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	result, err := podClient.Get(podName, metav1.GetOptions{})
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	framework.Logf("Created pod %q with IP %q %q", result.GetName(), result.Status.PodIP, result.Status.PodIPs)
+
+	err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+
+	if err != nil {
```

Comment:
shouldn´t we return here instead of continue and trying to delete the pod?

Reply:
Yup.Thanks for catching. Fixing in upcoming patch.